include/aspect/particle/world.h

```diff
@@ -21,729 +21,310 @@
-         * solution for this world.
+         * Const access to particles in this world.
          */
-        void set_solution(const LinearAlgebra::BlockVector *new_solution)
-        {
-          solution = new_solution;
-        };
+        const std::multimap<LevelInd, Particle<dim> > &
+        get_particles() const;
 
         /**
-         * Set the particle Integrator scheme for this particle world.
-         *
-         * @param [in] new_integrator The new Integrator scheme for this
-         * world.
+         * Advance particles by the old timestep using the current
+         * integration scheme. This accounts for the fact that the tracers
+         * are actually still at their old positions and the current timestep
+         * length is already updated for the next step at the time this
+         * function is called.
          */
-        void set_integrator(Integrator::Interface<dim, T> *new_integrator)
-        {
-          integrator = new_integrator;
-        };
+        void advance_timestep();
 
         /**
-         * Set the MPI communicator for this world.
+         * Calculates the global sum of particles over all processes. This is
+         * done to monitor the number of particles that have fallen out of the
+         * domain.
          *
-         * @param [in] new_comm_world The new MPI_Comm object for this world.
+         * @return Total number of particles in simulation.
          */
-        void set_mpi_comm(const MPI_Comm new_comm_world)
-        {
-          communicator = new_comm_world;
-
-          // Determine the size of the MPI comm world
-          world_size = Utilities::MPI::n_mpi_processes(communicator);
-          self_rank  = Utilities::MPI::this_mpi_process(communicator);
-        };
+        unsigned int get_global_particle_count() const;
 
         /**
-         * Get the MPI communicator associated with this particle world.
-         *
-         * @return associated MPI_Comm
+         * This callback function is called by Simulator to allow us to connect
+         * to the SimulatorSignals.
          */
-        MPI_Comm mpi_comm()
-        {
-          return communicator;
-        };
+        void
+        connector_function(aspect::SimulatorSignals<dim> &signals);
 
         /**
-         * All processes must call this function when finished adding
-         * particles to the world. This function will determine the total
-         * number of particles.
+         * Callback function that is called from Simulator before every
+         * refinement. Allows registering store_tracers() in the triangulation.
          */
-        void finished_adding_particles()
-        {
-          unsigned int local_num_particles = particles.size();
-
-          MPI_Allreduce(&local_num_particles, &global_num_particles, 1, MPI_UNSIGNED, MPI_SUM, communicator);
-        }
+        void
+        register_store_callback_function(typename parallel::distributed::Triangulation<dim> &triangulation);
```

Comment:
does this function register something, or does it get registered? In the latter case, the name is confusing. Same for the next function.